uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/common/DatePicker.java

```diff
@@ -238,12 +250,75 @@ public void setForceParse( final boolean forceParse ) {
      * Set the format of the Date shown in the TextBox component.
      * This is converted to BS3's Date Format that the underlying jQuery-based BS3 DatePicker
      * uses to convert values in the TextBox to selections in the popup date picker element.
-     * @param gwtDateFormat
+     * @param dateFormat
      */
-    public void setFormat( final String gwtDateFormat ) {
-        this.gwtDateFormat = gwtDateFormat;
+    public void setFormat( final String dateFormat ) {
+        setFormat(LocaleInfo.getCurrentLocale().getLocaleName(),dateFormat);
+    }
+
+    public void setFormat(String currentLang,  final String dateFormat) {
+        this.gwtDateFormat = setDatePickerLangProperties(currentLang, dateFormat);
         this.gwtDateTimeFormat = DateTimeFormat.getFormat( this.gwtDateFormat );
-        datePicker.setFormat( DatePickerFormatUtilities.convertToBS3DateFormat( gwtDateFormat ) );
+        datePicker.setFormat(DatePickerFormatUtilities.convertToBS3DateFormat(this.gwtDateFormat));
+    }
+
+    /**
+     * Sets the org.gwtbootstrap3.extras.datepicker.client.ui.DatePicker language and week start day
+     * depending on the activeLocale passed as parameter.
+     * There are incompatibilities between DateTimeFormatInfoImpl and DatePickerResourceBundles for es and de locales.
+     * This function modify in that cases the dateFormat to avoid show the name of the months
+     * TODO review locales es and de showing the name of the month.
```

Comment:
this have to be a temporary fix to the problems of the ES and DE with the month names. At least to could work with that locales. It's because of that I let here the TODO.